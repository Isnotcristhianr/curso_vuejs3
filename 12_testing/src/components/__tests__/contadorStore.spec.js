import { mount } from '@vue/test-utils'
import ContadorStore from '../contadorStore.vue'
import { describe, it } from 'vitest'
import { expect } from 'vitest'

describe('ContadorStore', () => {
    it('Debe renderizar el componente correctamente', () => {
        const wrapper = mount(ContadorStore)
        expect(wrapper.html()).toContain('Contador: 1')
    })

    it('Debe multiplicar el contador', async () => {
        const wrapper = mount(ContadorStore)
        await wrapper.find('button#multiply').trigger('click')
        expect(wrapper.find('h2').text()).toBe('Contador: 2')
    })

    it('Debe dividir el contador', async () => {
        const wrapper = mount(ContadorStore)
        await wrapper.find('button#divide').trigger('click')
        expect(wrapper.find('h2').text()).toBe('Contador: 1')
    })

    it('Debe reiniciar el contador', async () => {
        const wrapper = mount(ContadorStore)
        await wrapper.find('button#reset').trigger('click')
        await wrapper.find('button#reset').trigger('click')
        expect(wrapper.find('h2').text()).toBe('Contador: 0')
    })  
    
})

    