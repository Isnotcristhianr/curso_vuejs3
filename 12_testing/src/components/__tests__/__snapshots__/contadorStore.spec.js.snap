// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContadorStore > Debe renderizar el componente correctamente 1`] = `
"<div>
  <h2>Contador: 1</h2><button>Multiplicar</button><button>Dividir</button><button>Reiniciar</button>
</div>"
`;
